### PR TITLE
Cw340 hyperdebug

### PR DESCRIPTION
**Pull request by engdoreis**
This PR creates the interface hyper340 on OTT, and enable on test to run on a hyperdebug + cw340 kit.

### CODE REVIEW

**Review by pamaury — APPROVED** [review]
It looks good to me.

File: sw/host/opentitanlib/src/app/config/hyperdebug_cw310.json [hunk 1]
```diff
@@ -1,248 +1,12 @@
...
-      "alias_of": "CN7_9"
-    },
-    {
-      "name": "CC2",
-      "alias_of": "CN7_10"
-    }
-  ],
   "spi": [
     {
       "name": "BOOTSTRAP",
```
Comment (jesultra) [inline sw/host/opentitanlib/src/app/config/hyperdebug_cw310.json:6]: nit: If you wanted to, you could put a declaration mentioning "name", "chip_select", and "alias_of" in the common `hyperdebug_chipwhisperer.json`, and then have only "name" and "bits_per_sec" in each of the specific files.  This is fine as well.

File: sw/host/opentitanlib/src/transport/hyperdebug/mod.rs [hunk 2]
```diff
@@ -732,7 +734,7 @@ impl Flavor for ChipWhispererFlavor {
 
         // First, try to establish a connection to the native Chip Whisperer interface
         // which we will use for bitstream loading.
-        let board = ChipWhisperer::<Cw310>::new(None, None, None, &[])?;
+        let board = ChipWhisperer::<B>::new(None, None, None, &[])?;
```
Comment (jesultra) [inline sw/host/opentitanlib/src/transport/hyperdebug/mod.rs:737]: This is actually interesting.  In PR #19736 I am starting the work of making the `ChipWhisperer` transport not a Rust generic, but instead rely on `TransportWrapper` on top for doing all the remapping based on json files.  However, this in an instance where we use a ChipWhisperer instance without its own TransportWrapper.  No action necessary here, but I have to think about where the declarations of how the Sam3X should be used would come from (or maybe the bitstream loading is exactly the same between the two variants, if we are lucky.)

File: rules/opentitan_test.bzl [hunk 3]
```diff
@@ -420,6 +420,7 @@ def opentitan_functest(
                 },
                 "cw340": {
                     "cw340": "@//hw/bitstream/cw340",
+                    "hyper340": "@//hw/bitstream/cw340",
```
Comment (jesultra) [inline rules/opentitan_test.bzl:423]: I suppose the ribbon cable patch panel of CW340 enables us to use the exact same bitstream for both transports, unlike CW310?

**Review by jesultra — APPROVED** [review]

File: rules/opentitan_test.bzl (hunk 3 above)
Comment (engdoreis) [reply rules/opentitan_test.bzl:423]: Yes, for now we only have one bitstream.

File: sw/host/opentitanlib/src/transport/hyperdebug/mod.rs (hunk 2 above)
Comment (engdoreis) [reply sw/host/opentitanlib/src/transport/hyperdebug/mod.rs:737]: Yes, the bitstream loading is exactly the same in both boards, only the usb productId changes.

**Pull request merged**